apt_offline_core/AptOfflineCoreLib.py

```diff
@@ -872,8 +872,16 @@ def fetcher( args ):
         
         # get opts
         Str_GetArg = args.get
-        Int_SocketTimeout = args.socket_timeout
         Str_DownloadDir = args.download_dir
+        if not args.basepath is None:
+                if not os.path.isabs(args.get):
+                        Str_GetArg = os.path.join(args.basepath, os.path.basename(args.get))
+                if not os.path.isabs(args.download_dir):
+                        Str_DownloadDir = os.path.join(args.basepath, os.path.basename(args.download_dir))
+                        if not os.path.exists(Str_DownloadDir):
+                                os.mkdir(Str_DownloadDir, 777)
```

Comment:
I am not finding the --basepath idea of any use at all. Here, you still make an assumption that your basepath is fully writable, which isn't necessary. I'd rather prefer things to be explicit or just fail.


Quoting my previous comment (@github doesn't link/number comments):

```
[Hans] I'm happy to implement this error message, as well as
/etc/apt-offline.conf and --basepath if you agree to the approach.


[RRS] If we really want to, then just switching some of the non-mandatory
options to mandatory, will suffice the requirement. You can add the
defaults right in the argument parser.

Then the 2 together should be good, so that if the .conf file is non-
existent, it should still take sensible defaults.

```